cosmic-metrics-collector/src/main/java/com/github/missioncriticalcloud/cosmic/metricscollector/tasks/PublicIPAddressMetricCollector.java

```diff
@@ -0,0 +1,73 @@
+package com.github.missioncriticalcloud.cosmic.metricscollector.tasks;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.github.missioncriticalcloud.cosmic.metricscollector.exceptions.FailedToCollectMetricsException;
+import com.github.missioncriticalcloud.cosmic.metricscollector.model.Metric;
+import com.github.missioncriticalcloud.cosmic.metricscollector.repositories.PublicIPAddressMetricsRepository;
+import org.springframework.amqp.core.AmqpTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StopWatch;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+@Component
+public class PublicIPAddressMetricCollector implements MetricCollector{
```

Comment:
Please add a space between `MetricCollector` and `{` for better reading of the code.